### PR TITLE
Adding a React-less alternative import

### PR DESCRIPTION
**Pull request by rvaidya**
I'm taking a first stab at this here - I don't really like the naming I've used but this functions as a working prototype.  This also just sort of hacks in the functionality without trying to address bigger questions about package renaming, splitting, etc.

My use case is that I don't want React in my Webpack bundle, and importing react-ga forces React in.  Making React a peer/optional dependency doesn't really work because react-ga will just crash without it in the browser.  However, react-ga is the best Google Analytics wrapper I've found.

This pr adds another JS index file that can be imported instead of the main import.  You'd replace the default usage with:

`import ReactGA from 'react-ga/noreact';`

OutboundLink is not included in this import and cannot be used.

Related issues:
#205

### CODE REVIEW

**Comment by rvaidya** [comment]
Apologies for the errors, I'm developing on Windows and get tons of linting errors / tests aren't running.

File: noreact.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { default as default } from "./types/index";
```
Comment (SimeonC) [inline noreact.d.ts]: Shouldn't this be in the `/types` folder not the root?
Or does it need to be here for the exports, I'm assuming it's for something like `import ga from 'react-ga/noreact`;`

**Comment by rvaidya** [comment]
"core" does sound better, I can update the PR when I am able to.

Yes, the import needs to be next to the root source file with the same name
for the reason you stated, since "typings" only works for the default
import.

**Comment by DveMac** [comment]
I feel that this should probably be a refactor that extracts the library agnostic functionality to a separate (core) library, one which react-ga imports. Essentially making react-ga itself very small. react-ga could continue to expose the same API, so that is backwards compatible.

For example:
```
react-ga/es6-ga  (agnostic ga wrapper)
react-ga/react-ga  (react components that use es6-ga)
```

**Comment by SimeonC** [comment]
I think this will be fine as a stopgap - I'll check this locally this weekend and see if the build might break peoples current implementations (I don't think it will). If it may cause problems I think I'll just defer this to #311

**Comment by rvaidya** [comment]
About time I updated this

**Pull request merged**